library/core/src/any.rs

```diff
@@ -774,22 +809,14 @@ impl hash::Hash for TypeId {
         // - It is correct to do so -- only hashing a subset of `self` is still
         //   compatible with an `Eq` implementation that considers the entire
         //   value, as ours does.
-        self.t.1.hash(state);
+        self.data.full_hash[..8].hash(state);
```

Comment:
This no longer hashes a `u64` but instead a slice, which goes against the intended behavior here.
It's less efficient than just doing u64 due to the unnecessary length prefix and will break some code that relies on u64.
It will also do an unnecessary pointer read on 64-bit.
I suggest just using `partial_hash as u64` on 64-bit targets (or higher) and turning the first 8 bytes into a u64 on 32-bit and 16-bit targets.